pkg/steps/release/promote.go

```diff
@@ -176,7 +185,19 @@ func getImageMirrorTarget(tags map[string][]api.ImageStreamTagReference, pipelin
 		}
 		dockerImageReference = getPublicImageReference(dockerImageReference, pipeline.Status.PublicDockerImageRepository)
 		for _, dst := range dsts {
-			imageMirror[fmt.Sprintf("%s/%s", registry, dst.ISTagName())] = dockerImageReference
+			k := fmt.Sprintf("%s/%s", registry, dst.ISTagName())
+			if registry != api.QuayOpenShiftCIRepo {
+				imageMirror[k] = dockerImageReference
+			} else {
+				if quayTags, err := tagsInQuay(dockerImageReference, dst, date); err != nil {
+					logrus.WithField("key", k).WithError(err).
+						Warn("Failed to get the tag in quay.io and skipped the promotion to quay for this image")
+				} else {
+					for _, quayTag := range quayTags {
+						imageMirror[quayTag] = dockerImageReference
+					}
+				}
+			}
 			namespaces.Insert(dst.Namespace)
```

Comment:
Specifically, these two branches are exactly what the body of the two parameter functions should be.

Reply:
> I don't understand why we are back to explicitly testing step and registry names

I just feel it is easier this way than parsing function. But I will try another time the latter one.